app/controllers/oauth/applications_controller.rb

```diff
@@ -36,51 +38,61 @@ def create
 
       OSU::AccessPolicy.require_action_allowed!(:create, @user, @application)
 
-      if @application.save
-        security_log :application_created, application_id: @application.id,
-                                           application_name: @application.name
-        flash[:notice] = I18n.t(
-          :notice, scope: %i[doorkeeper flash applications create]
-        )
-        respond_to do |format|
-          format.html { redirect_to oauth_application_url(@application) }
-          format.json { render json: @application }
-        end
-      else
-        respond_to do |format|
-          format.html { render :new }
-          format.json do
-            render json: { errors: @application.errors.full_messages },
-                   status: :unprocessable_entity
+      Doorkeeper::Application.transaction do
+        if handle_member_ids(params["member_ids"]) && @application.save
+          @application.owner.save
+          security_log :application_created, application_id: @application.id,
+                                            application_name: @application.name
+          flash[:notice] = I18n.t(
+            :notice, scope: %i[doorkeeper flash applications create]
+          )
+          respond_to do |format|
+            format.html { redirect_to oauth_application_url(@application) }
+            format.json { render json: @application }
+          end
+        else
+          respond_to do |format|
+            format.html { render :new }
+            format.json do
+              render json: { errors: @application.errors.full_messages },
+                    status: :unprocessable_entity
+            end
           end
+          raise ActiveRecord::Rollback
         end
       end
     end
 
     def edit
       OSU::AccessPolicy.require_action_allowed!(:update, @user, @application)
+      @member_ids = set_member_ids
     end
 
     def update
       OSU::AccessPolicy.require_action_allowed!(:update, @user, @application)
-
-      if @application.update_attributes(app_params)
-        security_log :application_updated, application_id: @application.id,
-                                           application_params: app_params
-        flash[:notice] = I18n.t(
-          :notice, scope: %i[doorkeeper flash applications update]
-        )
-        respond_to do |format|
-          format.html { redirect_to oauth_application_url(@application) }
-          format.json { render json: @application }
-        end
-      else
-        respond_to do |format|
-          format.html { render :edit }
-          format.json do
-            render json: { errors: @application.errors.full_messages },
-                   status: :unprocessable_entity
+      
+      Doorkeeper::Application.transaction do
+        if handle_member_ids(params["member_ids"]) && @application.update_attributes(app_params)
```

Comment:
I think you should be using `app_params[:member_ids]` (or similar) because `params` hasn't been permitted... when you call `.permit` on `params` down in `def app_params`, it is [returning a new object, not modifying the `params` object](https://apidock.com/rails/ActionController/Parameters/permit).

Reply:
This is fixed